packages/lisk-p2p/src/sanitization.ts

```diff
@@ -92,3 +98,32 @@ export const sanitizePeerInfoList = (
 		throw new InvalidRPCResponse('Invalid response type');
 	}
 };
+
+export const validateRPCRequest = (request: unknown): P2PRequest => {
+	if (!request) {
+		throw new InvalidRPCRequestError('Invalid request');
+	}
+
+	const rpcRequest = request as P2PRequest;
```

Comment:
We can't cast the `request` object into a `P2PRequest ` because `P2PRequest` is a usable request object which can be responded to using its `request.end(...)` or `request.error(...)` method. Maybe this validation/sanitization function should just return a `ProtocolRPCRequestPacket`; that's also more consistent with the `sanitizeProtocolMessage` function below. We will have to instantiate the `P2PRequest` as a separate step (not part of validation/sanitization); it needs the SocketCluster `respond` callback as third argument to its constructor.